src/lib/plugin.js

```diff
@@ -934,4 +937,34 @@ function * requestRetry (requestOptions, retryOptions) {
   throw new Error(retryOptions.errorMessage)
 }
 
+function setupPing (ws) {
+  let isAlive = true
+  let pingInterval
+
+  function clear () {
+    clearInterval(pingInterval)
+    ws.removeListener('pong', heartbeat)
+  }
+
+  function heartbeat () {
+    isAlive = true
+  }
+
+  function ping () {
+    if (!isAlive) clear()
```

Comment:
Should this return? `ping()`-ing again won't do much if the `'pong'` event listener was just removed.